### PR TITLE
Farmback 04 init passport

### PR DESCRIPTION
**Pull request by Erriep**

### CODE REVIEW

**Review by AntoineCAU — APPROVED** [review]

**Comment by AntoineCAU** [comment]
Nice job Pierre !

File: passport-strategies.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        );
+        if (!sqlRes.length) return done(null, false);
+        const { id, email, firstname, lastname, password } = sqlRes[0];
+        const isPasswordOK = bcrypt.compareSync(formPassword, password);
+        if (!isPasswordOK) return done(null, false, 'Wrong password!');
+
+        const user = { id, email, firstname, lastname };
+        return done(null, user);
+      } catch (e) {
+        console.log(e);
```
Comment (LoicBrassart) [inline passport-strategies.js]: No `console.log`, delete it

File: routes/auth.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
+const express = require('express');
+const passport = require('passport');
+
```
Comment (LoicBrassart) [inline routes/auth.js:3]: useless empty line

File: routes/auth.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  try {
+    const formData = req.body;
+    formData.password = bcrypt.hashSync(formData.password, jwtrounds);
+    const [sqlRes] = await db.query(`INSERT INTO users SET ?`, formData);
+    delete formData.password;
+    formData.id = sqlRes.insertId;
+    const token = jwt.sign(formData, jwtsecret);
+    res.status(201).json(token);
+  } catch (e) {
+    console.log(e);
```
Comment (LoicBrassart) [inline routes/auth.js]: no console.log

File: routes/misc.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
+const express = require('express');
+const passport = require('passport');
+
```
Comment (LoicBrassart) [inline routes/misc.js]: useless empty line

File: routes/misc.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+});
+
+router.get('/testDB', async (req, res) => {
+  try {
+    const [sqlRes] = await db.execute(`SELECT * FROM user`);
+    res.status(200).json(sqlRes);
+  } catch (e) {
+    res.status(500).json(e);
+  }
+});
```
Comment (LoicBrassart) [inline routes/misc.js]: Useless routes

**Review by LoicBrassart — CHANGES_REQUESTED** [review]
Code to remove

File: routes/auth.js (hunk 2 above)
Comment (Erriep) [reply routes/auth.js:3]: es-lint force this :(

File: routes/misc.js (hunk 4 above)
Comment (Erriep) [reply routes/misc.js]: es-lint force this :(

**Review by AntoineCAU — APPROVED** [review]

File: routes/misc.js [hunk 6]
```diff
@@ -0,0 +1,11 @@
+const express = require('express');
+const passport = require('passport');
+
+const router = express.Router();
+
+router.get('/protected', passport.authenticate('jwt'), (req, res) => {
+  const msg = `If you can see this, you should be logged in, ${req.user.firstname}`;
+  res.status(200).send(msg);
+});
```
Comment (LoicBrassart) [inline routes/misc.js]: Useless route on this project

**Review by LoicBrassart — CHANGES_REQUESTED** [review]
One route to delete

**Review by AntoineCAU — APPROVED** [review]

**Review by LoicBrassart — APPROVED** [review]
Better!

**Pull request merged**